diff --git a/ble-host/sources/stack/smp/smp_main.c b/ble-host/sources/stack/smp/smp_main.c
index ba48893..0000000 100644
--- a/ble-host/sources/stack/smp/smp_main.c
+++ b/ble-host/sources/stack/smp/smp_main.c
@@ -37,6 +37,8 @@
 #include "smp_handler.h"
 #include "sec_api.h"
 
+#include "sec_main.h"
+#include "wsf_os_int.h"
 /**************************************************************************************************
   Global Variables
 **************************************************************************************************/
@@ -872,7 +874,16 @@ void SmpHandler(wsfEventMask_t event, wsfMsgHdr_t *pMsg)
         /* if AES result verify it is not stale */
         if (pMsg->event == SMP_MSG_WSF_AES_CMPL && pCcb->token != pMsg->status)
         {
-          SMP_TRACE_WARN2("AES token mismatch: %d %d", pCcb->token, pMsg->status);
+            extern secCb_t secCb;
+            wsfHandlerId_t  handlerId = 0;
+            secQueueBuf_t *pBuf = NULL;
+
+            SMP_TRACE_WARN2("AES token mismatch: %d %d", pCcb->token, pMsg->status);
+
+            while((pBuf=WsfMsgDeq(&secCb.aesEncQueue, &handlerId))!=NULL)
+            {
+                WsfMsgFree(pBuf);
+            }
         }
         else
         {
