--- SDK//CMSIS/AmbiqMicro/Source/system_apollo510.c

+++ DFP/Device/Source/system_apollo510.c

@@ -40,7 +40,7 @@

 // ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
 // POSSIBILITY OF SUCH DAMAGE.
 //
-// This is part of revision release_sdk5p2p0-66487dd10 of the AmbiqSuite Development Package.
+// This is part of the Ambiq CMSIS Pack for the AmbiqSuite Development Package.
 //
 //*****************************************************************************
 
