--- a/utility/libc/tinyprintf.c
+++ b/utility/libc/tinyprintf.c
@@ -188,8 +188,10 @@
 
     /* Fill with space, before alternate or sign */
     if (!p->lz) {
-        while (n-- > 0)
+        while (n > 0) {
+            --n;
             written += putf(putp, ' ');
+        }
     }
 
     /* print sign */
@@ -206,8 +208,10 @@
 
     /* Fill with zeros, after alternate or sign */
     if (p->lz) {
-        while (n-- > 0)
+        while (n > 0) {
+            --n;
             written += putf(putp, '0');
+        }
     }
 
     /* Put actual buffer */
