--- a/e_fmod.c
+++ b/e_fmod.c
@@ -22,9 +22,9 @@
 #ifndef _DOUBLE_IS_32BITS
 
 #ifdef __STDC__
-static const double one = 1.0, Zero[] = {0.0, -0.0,};
+static const double one = 1.0;
 #else
-static double one = 1.0, Zero[] = {0.0, -0.0,};
+static double one = 1.0;
 #endif
 
 #ifdef __STDC__
@@ -50,7 +50,7 @@
 	if(hx<=hy) {
 	    if((hx<hy)||(lx<ly)) return x;	/* |x|<|y| return x */
 	    if(lx==ly) 
-		return Zero[(__uint32_t)sx>>31];	/* |x|=|y| return x*0*/
+		{ INSERT_WORDS(x,sx,0); return x; }	/* |x|=|y| return x*0*/
 	}
 
     /* determine ix = ilogb(x) */
@@ -104,7 +104,7 @@
 	    if(hz<0){hx = hx+hx+(lx>>31); lx = lx+lx;}
 	    else {
 	    	if((hz|lz)==0) 		/* return sign(x)*0 */
-		    return Zero[(__uint32_t)sx>>31];
+		    { INSERT_WORDS(x,sx,0); return x; }
 	    	hx = hz+hz+(lz>>31); lx = lz+lz;
 	    }
 	}
@@ -113,7 +113,7 @@
 
     /* convert back to floating value and restore the sign */
 	if((hx|lx)==0) 			/* return sign(x)*0 */
-	    return Zero[(__uint32_t)sx>>31];	
+	    { INSERT_WORDS(x,sx,0); return x; }	
 	while(hx<0x00100000) {		/* normalize x */
 	    hx = hx+hx+(lx>>31); lx = lx+lx;
 	    iy -= 1;
