diff --git a/third-party/upstream/nationalchip-lvp-kws/lvp/common/uart_message_v2.c b/g2/analysis/source-discovery-parallel-2026-10-09/acquisitions/nationalchip-lvp-aiot/lvp/common/uart_message_v2.c
index b5ae5ebac..fd96127ae 100644
--- a/third-party/upstream/nationalchip-lvp-kws/lvp/common/uart_message_v2.c
+++ b/g2/analysis/source-discovery-parallel-2026-10-09/acquisitions/nationalchip-lvp-aiot/lvp/common/uart_message_v2.c
@@ -1,5 +1,5 @@
 /* Voice Signal Preprocess
- * Copyright (C) 2001-2020 NationalChip Co., Ltd
+ * Copyright (C) 2001-2023 NationalChip Co., Ltd
  * ALL RIGHTS RESERVED!
  *
  * uart_message_v2.c: MCU uart communication protocol
